pkg/cri/server/helpers.go

```diff
@@ -523,10 +523,8 @@ func copyResourcesToStatus(spec *runtimespec.Spec, status containerstore.Status)
 func (c *criService) generateAndSendContainerEvent(ctx context.Context, containerID string, sandboxID string, eventType runtime.ContainerEventType) {
 	podSandboxStatus, err := c.getPodSandboxStatus(ctx, sandboxID)
 	if err != nil {
-		// TODO(https://github.com/containerd/containerd/issues/7785):
-		// Do not skip events with nil PodSandboxStatus.
-		logrus.Errorf("Failed to get podSandbox status for container event for sandboxID %q: %v. Skipping sending the event.", sandboxID, err)
-		return
+		logrus.Errorf("Failed to get podSandbox status for container event for sandboxID %q: %v. Sending the event with podSandboxStatus being nil.", sandboxID, err)
+		podSandboxStatus = nil
```

Comment:
Nit: this line seems to be not needed. 
If err!=nil, podSandboxStatus is already nil.

Reply:
yep that's true with current implementation. I am explicitly setting it nil (again) here just in case the implementation of getPodSandboxStatus changes in future.

Reply:
I think this is a reasonable defensive-programming approach right now, since we explicitly want to ensure that podSandboxStatus remains nil in this case (and it's tied directly to the log line immediately above).